src/transaction_builder.js

```diff
@@ -436,29 +446,36 @@ TransactionBuilder.prototype.__build = function (allowIncomplete) {
   return tx
 }
 
+function canSign (input) {
+  return input.hashType !== undefined &&
+    input.prevOutScript !== undefined &&
+    input.pubKeys !== undefined &&
+    input.signatures !== undefined &&
+    input.signatures.length === input.pubKeys.length &&
```

Comment:
@afk11 yes,  the number of signatures required is not `===` to the number of possible signatures,  which is what `.signatures` represents.

E.g,  for P2PKH,  before signing,  `signatures === [undefined]`